pkg/config/config.go

```diff
@@ -48,27 +48,33 @@ func LoadConf(bytes []byte) (*sriovtypes.NetConf, error) {
 	}
 
 	if hostIFNames != "" {
-		n.HostIFNames = hostIFNames
+		n.OrigVfState.HostIFName = hostIFNames
 	}
 
 	if hostIFNames == "" && !n.DPDKMode {
 		return nil, fmt.Errorf("LoadConf(): the VF %s does not have a interface name or a dpdk driver", n.DeviceID)
 	}
 
-	// validate vlan id range
-	if n.Vlan < 0 || n.Vlan > 4094 {
-		return nil, fmt.Errorf("LoadConf(): vlan id %d invalid: value must be in the range 0-4094", n.Vlan)
+	if n.Vlan != nil {
+		// validate vlan id range
+		if *n.Vlan < 0 || *n.Vlan > 4094 {
+			return nil, fmt.Errorf("LoadConf(): vlan id %d invalid: value must be in the range 0-4094", *n.Vlan)
+		}
+	}
+
+	if n.VlanQoS != nil {
+		// validate that VLAN QoS is in the 0-7 range
+		if *n.VlanQoS < 0 || *n.VlanQoS > 7 {
+			return nil, fmt.Errorf("LoadConf(): vlan QoS PCP %d invalid: value must be in the range 0-7", *n.VlanQoS)
+		}
 	}
 
 	// validate that vlan id is set if vlan qos is configured
-	if n.Vlan == 0 && n.VlanQoS != 0 {
+	if (n.Vlan == nil || *n.Vlan == 0) && n.VlanQoS != nil {
```

Comment:
I know it's rarely used, but shall we allow combined settings `vlan == 0` and `vlanQoS == 0`?

Reply:
IMO we should not (at least not in this PR) ill modify the condition to:

`if (n.Vlan == nil || *n.Vlan == 0) && (n.VlanQos != nil && *n.VlanQos != 0) {...}`

Reply:
On second thought, i don't think we need to prevent it.

Changing Vlan and VlanQos to a pointer allows us to know if the user specified a value.
so in my opinion we should enforce:

1. Vlan must be specified by the user if VlanQos is specified
2. if VlanQos is specified and is a non-zero value then Vlan must be a non zero value

i don't think we should prevent `vlan == 0 && vlanQos == 0`